project/src/main/java/com/google/sps/servlets/VolunteeringFormHandlerServlet.java

```diff
@@ -14,34 +14,57 @@
 /** Servlet that handles submission of the form for creating volunteering opportunities. */
 @WebServlet("/volunteering-form-handler")
 public class VolunteeringFormHandlerServlet extends HttpServlet {
+  private static final String OPPORTUNITY_ID = "opportunity-id";
   private static final String OPPORTUNITY_NAME = "opportunity-name";
   private static final String OPPORTUNITY_NUM_SPOTS = "opportunity-num-spots";
   private static final String SKILL = "skill";
   private static final String HARDCODED_EVENT_ID = "0883de79-17d7-49a3-a866-dbd5135062a8";
+  private static final DatabaseWrapper databaseWrapper =
+      new DatabaseWrapper(DatabaseConstants.INSTANCE_ID, DatabaseConstants.DATABASE_ID);
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String volunteerName =
+    String opportunityId = request.getParameter(OPPORTUNITY_ID);
+    String opportunityName =
         CommonUtils.getParameter(request, OPPORTUNITY_NAME, /* DefaultValue= */ "");
-    Integer volunteerNumber =
+    int opportunityNumberOfSpots =
         Integer.parseInt(
             CommonUtils.getParameter(request, OPPORTUNITY_NUM_SPOTS, /* DefaultValue= */ "0"));
     Set<String> skills = CommonUtils.getParameterValues(request, SKILL);
 
-    insertVolunteeringOpportunityIntoDB(volunteerName, volunteerNumber, skills);
+    // If opportunityId is not passed as a parameter, perform an insert else perform an update
+    if (opportunityId == null)
+      insertVolunteeringOpportunityInDB(opportunityName, opportunityNumberOfSpots, skills);
+    else
+      updateVolunteeringOpportunityInDB(
+          opportunityId, opportunityName, opportunityNumberOfSpots, skills);
 
     response.sendRedirect("/events-feed.html");
   }
 
-  private static void insertVolunteeringOpportunityIntoDB(
-      String volunteerName, Integer volunteerNumber, Set<String> requiredSkills) {
-    DatabaseWrapper dbWrapper =
-        new DatabaseWrapper(DatabaseConstants.INSTANCE_ID, DatabaseConstants.DATABASE_ID);
+  private static void insertVolunteeringOpportunityInDB(
+      String opportunityName, int opportunityNumberOfSpots, Set<String> requiredSkills) {
```

Comment:
Ditto, opportunityRequiredSkills if it's only used for opportunities for consistency

Reply:
Done, thanks - made naming consistent.